bentoctl_sagemaker/create_deployable.py

```diff
@@ -58,22 +37,43 @@ def create_deployable(
 
     Returns
     -------
-    dockerfile_path : str
-        path to the dockerfile.
     docker_context_path : str
         path to the docker context.
-    additional_build_args : dict
-        Any addition build arguments that need to be passed to the
-        docker build command
     """
-    deployable_path = os.path.join(destination_dir, "bentoctl_deployable")
-    docker_context_path = deployable_path
+    deployable_path = Path(destination_dir)
 
     # copy over the bento bundle
-    shutil.copytree(bento_path, deployable_path)
-    dockerfile_path = generate_dockerfile_in(deployable_path, bento_metadata)
-    generate_sagemaker_service_in(deployable_path)
-    generate_serve_script_in(deployable_path)
+    shutil.copytree(bento_path, deployable_path, dirs_exist_ok=True)
+
+    bento_metafile = Path(bento_path, "bento.yaml")
+    with bento_metafile.open("r", encoding="utf-8") as metafile:
+        info = BentoInfo.from_yaml_file(metafile)
+
+    options = bentoml_cattr.unstructure(info.docker)
+    options["dockerfile_template"] = TEMPLATE_PATH
+
+    dockerfile_path = deployable_path.joinpath("env", "docker", "Dockerfile")
+    with dockerfile_path.open("w", encoding="utf-8") as dockerfile:
+        dockerfile.write(
+            generate_dockerfile(
+                DockerOptions(**options).with_defaults(),
+                str(deployable_path),
+                use_conda=any(
+                    i is not None
+                    for i in bentoml_cattr.unstructure(info.conda).values()
+                ),
+            )
+        )
```

Comment:
throw error:
```
Traceback (most recent call last):
  File "/usr/local/Caskroom/miniconda/base/envs/dev/bin/bentoctl", line 5, in <module>
    bentoctl()
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/cli/utils.py", line 90, in wrapper
    return_value = func(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/cli/utils.py", line 55, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/cli/utils.py", line 24, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/cli/__init__.py", line 153, in build
    destination_dir=dist_dir,
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/deployment_config.py", line 218, in create_deployable
    overwrite_deployable=overwrite_deployable,
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/operator/operator.py", line 128, in create_deployable
    bento_path, destination_dir, bento_metadata, overwrite_deployable
  File "/Users/bozhaoyu/src/aws-sagemaker-deploy/bentoctl_sagemaker/create_deployable.py", line 68, in create_deployable
    for i in bentoml_cattr.unstructure(info.conda).values()
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoml/_internal/bento/gen.py", line 214, in generate_dockerfile
    return template.render(**get_templates_variables(options, use_conda=use_conda))
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/bozhaoyu/src/aws-sagemaker-deploy/bentoctl_sagemaker/sagemaker/template.j2", line 1, in top-level template code
    {% extends bento__dockerfile %}
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/jinja2/loaders.py", line 195, in get_source
    pieces = split_template_path(template)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/jinja2/loaders.py", line 30, in split_template_path
    for piece in template.split("/"):
jinja2.exceptions.UndefinedError: 'bento__dockerfile' is undefined
```

Reply:
we need an update on the extends field. I will fix that later td.

Reply:
that fix is on sagemaker or on rc1?

Reply:
sounds good. keep me posted. Feel free to tag along the fix for other two issues I mentioned.